Components/MineSharp.Protocol/MinecraftClient.cs

```diff
@@ -567,13 +754,17 @@ private async Task HandleOutgoingPacket(IPacket packet)
             client.Dispose();
```

Comment:
changing this to `await client.DisposeAsync();` fixed the following exception:

2024/08/14 23:06:13.901|Warn|MineSharp.Protocol.MinecraftClient|An packet handler for packet of type CB_Status_ServerInfo threw an exception. System.Threading.Tasks.TaskCanceledException: A task was canceled.
at System.Threading.Tasks.Task.GetExceptions(Boolean includeTaskCanceledExceptions)
at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
at System.Threading.Tasks.Task.Wait()
at MineSharp.Protocol.MinecraftClient.Dispose() in /home/psu/projects/MineSharp/Components/MineSharp.Protocol/MinecraftClient.cs:line 221
at MineSharp.Protocol.MinecraftClient.<>c__DisplayClass56_0.<<RequestServerStatus>b__0>d.MoveNext() in /home/psu/projects/MineSharp/Components/MineSharp.Protocol/MinecraftClient.cs:line 754
[...]
at MineSharp.Core.Concurrency.ConcurrencyHelper.<>c__DisplayClass0_0.<<EnsureOnlyRunOnceAsync>b__0>d.MoveNext() in /home/psu/projects/MineSharp/MineSharp.Core/Concurrency/ConcurrencyHelper.cs:line 27
[...]
at MineSharp.Protocol.MinecraftClient.DisconnectInternal(Chat reason) in /home/psu/projects/MineSharp/Components/MineSharp.Protocol/MinecraftClient.cs:line 323

Reply:
Ok the actual problem is that client.Disconnect() should not be called here. No point in disconnecting when the server will close the connection anyways. There should only be `await client.DisposeAsync();`.
I make a commit with that